modules/mysql.lua

```diff
@@ -6,35 +6,44 @@
 local l = require "lpeg"
 l.locale(l)
 local tonumber = tonumber
+local ip = require "ip_address"
 
 local M = {}
 setfenv(1, M) -- Remove external access to contain everything in the module
 
+local pct_encoded   = l.P"%" * l.xdigit * l.xdigit
+local unreserved    = l.alnum + l.S"-._~"
+local sub_delims    = l.S"!$&'()*+,;="
+
 local space         = l.space^1
 local sep           = l.P"\n"
 local sql_end       = l.P";" * (l.P"\n" + -1)
 local line          = (l.P(1) - sep)^0 * sep
-local float         = l.digit^1 * "." * l.digit^1
+local float         = l.digit^1 * "." * l.digit^1 / tonumber
+local integer       = l.P"-"^-1 * l.digit^1 / tonumber
+local host          = l.Ct(l.Cg(ip.v4, "value") * l.Cg(l.Cc"ipv4", "representation"))
+                    + l.Ct(l.Cg(ip.v6, "value") * l.Cg(l.Cc"ipv6", "representation"))
+                    + l.Ct(l.Cg((unreserved + pct_encoded + sub_delims)^1, "value") * l.Cg(l.Cc"hostname", "representation"))
```

Comment:
We now have something like this is multiple places.  We should probably pull host/hostname out into a common grammar module like ip_address.lua (not necessary for this PR)